src/components/Main/Main.js

```diff
@@ -0,0 +1,393 @@
+    const audio = document.querySelector('.game-music');
+    audio.loop = true;
+    if (!this.state.isMusicOn) {
+      audio.volume = this.state.musicVolume;
+      audio.play();
+    } else {
+      audio.pause();
+      audio.currentTime = 0;
+    }
+    this.setState((state) => {
+      return {
+        isMusicOn: !state.isMusicOn,
+      };
+    });
+  };
+
+  onSoundSwitcherClick = () => {
+    this.setState((state) => {
+      return {
+        isSoundOn: !state.isSoundOn,
+      };
+    });
+  };
+
+  isGuessd = () => {
+    if (this.firstTurnedCard && this.secondTurnedCard) {
+      return this.firstTurnedCard.setId === this.secondTurnedCard.setId;
+    }
+    return false;
+  };
+
+  turnTwoCards = (firstCardId, secondCardId) => {
+    this.setState((state) => {
+      const newState = { ...state };
+      newState.cardSet.find((cardObj) => cardObj.id === firstCardId).isTurned = false;
+      newState.cardSet.find((cardObj) => cardObj.id === secondCardId).isTurned = false;
+      newState.moveCounter++;
+      return newState;
+    });
+  };
+
+  addMove = () => {
+    this.setState((state) => {
+      return {
+        moveCounter: state.moveCounter + 1,
+      };
+    });
+  };
+
+  onCardClick = (id) => {
+
+    if (this.state.isSoundOn) {
+      const audio = new Audio(cardTurnSound);
+      audio.volume = this.state.soundsVolume;
+      audio.play();
+    }
+
+    const pickedCardObj = this.state.cardSet.find((cardObj) => cardObj.id === id);
+    const pickedCardObjCoppy = { ...pickedCardObj };
+
+    if (pickedCardObj.isGuessed) {
+      return;
+    }
+
+    this.setState((state) => {
+      const pickedCardObjIndex = state.cardSet.indexOf(pickedCardObj);
+      const newState = { ...state.cardSet };
+      newState[pickedCardObjIndex].isTurned = true;
+      pickedCardObjCoppy.isTurned = true;
+      return newState;
+    });
+
+    if (!this.firstTurnedCard) {
+      this.firstTurnedCard = pickedCardObjCoppy;
+    } else {
+      this.secondTurnedCard = pickedCardObjCoppy;
+
+      if (this.isGuessd()) {
+        setTimeout(this.addMove, 1000);
+        const firstTurnedCardId = this.firstTurnedCard.id;
+        const secondTurnedCardId = this.secondTurnedCard.id;
+        this.firstTurnedCard = null;
+        this.secondTurnedCard = null;
+        this.setState((state) => {
+          const newState = { ...state };
+          newState.cardSet.find((cardObj) => cardObj.id === firstTurnedCardId).isGuessed = true;
+          newState.cardSet.find((cardObj) => cardObj.id === secondTurnedCardId).isGuessed = true;
+          
+          if (this.isPlayerWon()) {
+            this.isGameFinish = true;
+            let rating;
+            if (this.state.cardsQuantity === '18') {
+              rating = JSON.parse(localStorage.getItem('ratingCardsQuantity18'));
+            } else {
+              rating = JSON.parse(localStorage.getItem('ratingCardsQuantity12'));
+            }
+            if (rating.length < 10) {
```

Comment:
just can't understand what are you doing in this `if` block - this is very weird and complicated